numpy/core/tests/test_multiarray.py

```diff
@@ -3549,8 +3551,18 @@ def test_test_zero_rank(self):
 
 
 class TestPickling(object):
+    def test_highest_available_pickle_protocol(self):
```

Comment:
This test is a bit strange, why is it needed?

Reply:
By the time I started writing this set of PRs, it was a sort of meta-test that made sure all pickle protocol test loops were actually including protocol 5 if possible (e.g if pickle5 was installed), especially for the `ndarray` class.
Now things have a bit changed: we loop over protocols in plenty of other test files, yet this test is only ran here. I agree that this test is less relevant now, and it can actually be removed for the sake of clarity and consistency.